handler.go

```diff
@@ -1495,3 +1498,134 @@ func errorString(err error) string {
 	}
 	return err.Error()
 }
+
+// handlePOSTInputDefinition handles POST /input-definition request.
+func (h *Handler) handlePostInputDefinition(w http.ResponseWriter, r *http.Request) {
+	indexName := mux.Vars(r)["index"]
+	inputDefName := mux.Vars(r)["input-definition"]
+
+	// Decode request.
+	var req InputDefinitionInfo
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Find index.
+	index := h.Holder.Index(indexName)
+	if index == nil {
+		http.Error(w, ErrIndexNotFound.Error(), http.StatusNotFound)
+		return
+	}
+
+	def, err := req.Encode()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	def.Name = inputDefName
+
+	// Validate columnLabel & duplicate primaryKey
+	numPrimaryKey := 0
+	for _, field := range def.Fields {
+		if field.PrimaryKey {
+			numPrimaryKey += 1
+			if field.Name == index.columnLabel {
+				continue
+			} else {
+				err = fmt.Errorf("primary field's name not match columnLabel")
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+		}
+	}
+	if numPrimaryKey > 1 {
+		err = errors.New("duplicate primaryKey with other field")
```

Comment:
let's make these errors part of the system error constants.
that way, when we check for them in tests, we don't have to compare the body of the error.